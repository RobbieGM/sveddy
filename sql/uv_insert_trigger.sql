DROP SCHEMA public CASCADE;
CREATE SCHEMA public;
GRANT ALL ON SCHEMA public TO current_user;
GRANT ALL ON SCHEMA public TO public;
CREATE EXTENSION sveddy;

CREATE TABLE ratings (
	uid integer,
	iid integer,
	rating integer
);
INSERT INTO ratings (uid, iid, rating) VALUES
(1, 1, 4),
(1, 2, 3),
(2, 1, 5),
(3, 3, 4);

CALL initialize_model_uv('ratings', 'uid', 'iid', 'rating', 2);

INSERT INTO ratings (uid, iid, rating) VALUES
(4, 4, 5);

SELECT id, cardinality(weights) FROM ratings_sveddy_model_u;
SELECT id, cardinality(weights) FROM ratings_sveddy_model_v;

INSERT INTO ratings (uid, iid, rating) VALUES
(4, 5, 4);

SELECT id, cardinality(weights) FROM ratings_sveddy_model_u;
SELECT id, cardinality(weights) FROM ratings_sveddy_model_v;
