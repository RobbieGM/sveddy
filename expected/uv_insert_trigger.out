DROP SCHEMA public CASCADE;
NOTICE:  drop cascades to 4 other objects
DETAIL:  drop cascades to extension sveddy
drop cascades to table ratings
drop cascades to table ratings_sveddy_model_u
drop cascades to table ratings_sveddy_model_v
CREATE SCHEMA public;
GRANT ALL ON SCHEMA public TO current_user;
GRANT ALL ON SCHEMA public TO public;
CREATE EXTENSION sveddy;
CREATE TABLE ratings (
	uid integer,
	iid integer,
	rating integer
);
INSERT INTO ratings (uid, iid, rating) VALUES
(1, 1, 4),
(1, 2, 3),
(2, 1, 5),
(3, 3, 4);
CALL initialize_model_uv('ratings', 'uid', 'iid', 'rating', 2);
INSERT INTO ratings (uid, iid, rating) VALUES
(4, 4, 5);
SELECT id, cardinality(weights) FROM ratings_sveddy_model_u;
 id | cardinality 
----+-------------
  3 |           2
  2 |           2
  1 |           2
  4 |           2
(4 rows)

SELECT id, cardinality(weights) FROM ratings_sveddy_model_v;
 id | cardinality 
----+-------------
  3 |           2
  2 |           2
  1 |           2
  4 |           2
(4 rows)

INSERT INTO ratings (uid, iid, rating) VALUES
(4, 5, 4);
SELECT id, cardinality(weights) FROM ratings_sveddy_model_u;
 id | cardinality 
----+-------------
  3 |           2
  2 |           2
  1 |           2
  4 |           2
(4 rows)

SELECT id, cardinality(weights) FROM ratings_sveddy_model_v;
 id | cardinality 
----+-------------
  3 |           2
  2 |           2
  1 |           2
  4 |           2
  5 |           2
(5 rows)

